Make text on taskbar readable on light color scheme

pbabinca opened the pull request:
Gnome 42+ supports setting color preferences: light (default) and dark[^1].

When I set preferred color scheme to default or prefer light - either in the Settings or with `gsettings set org.gnome.desktop.interface color-scheme default` or `gsettings set org.gnome.desktop.interface color-scheme prefer-light`, text on the taskbar stays slightly dark grey on light grey background which makes text unreadable:

![Screenshot from 2024-06-21 15-20-54](https://github.com/home-sweet-gnome/dash-to-panel/assets/2543518/c1beff47-7c06-4d45-9e71-976d935b3d54)

I'm running Fedora 40, Gnome Shell 46, using the default Adwaita theme.

After this pull request, there are two new settings to set font color of the application titles on light color scheme for minimized and not minimized applications in Dash to panel settings -> Behavior -> Ungroup applications -> settings icon:

![Screenshot from 2024-06-21 15-50-18](https://github.com/home-sweet-gnome/dash-to-panel/assets/2543518/3484e28d-f121-4ba7-bd4f-dc149358cafc)

By default this color is `#282828` which is Gnome Shell's dark text on light color scheme[^2] for `.panel-button`:
![Screenshot from 2024-06-21 15-51-25](https://github.com/home-sweet-gnome/dash-to-panel/assets/2543518/bc7c6dce-86c0-4371-a55a-16ff6a0ce9c6)


[^1]: Some information can be found on [Gnome Initiatives Wiki: Dark Style Preference](https://gitlab.gnome.org/GNOME/Initiatives/-/wikis/Dark-Style-Preference).

[^2]: Since Gnome shell uses Sass to build up css and then gresource, the easiest is to query it like this:
    ```
    gresource extract \
    /usr/share/gnome-shell/gnome-shell-theme.gresource \
    /org/gnome/shell/theme/gnome-shell-light.css \
    | awk '/#panel \.panel-button \{/,/\}/ { if ($0 ~ /color:/ && $0 !~ /background-color:/) print }'
    ```

julienfalque commented:
Instead of setting it, can't the color be automatically inherited from the shell somehow? This already works for e.g. the clock or the battery percentage.

Note: this relates to #1978.

pbabinca commented:
> Instead of setting it, can't the color be automatically inherited from the shell somehow? This already works for e.g. the clock or the battery percentage.

I don't know if it is possible to automatically _inherit the color_ as you are suggesting.

Dash to panel does not use CSS classes but rather sets styles directly based on other settings. See e.g. here:

https://github.com/home-sweet-gnome/dash-to-panel/blob/77ce0879f079b36676d38701e415927f4c3d6a17/appIcons.js#L587-L590

With my approach in this PR I ended up with a fairly small change in the code base and environment that I'm not familiar with. I'm using this for some time now, it does not causes me any issues and I'm fine with this solution.

pbabinca committed:
Color scheme-aware text color on taskbar

Automatically uses text with dark color when theme prefers light color,
or user uses light-style extension. Both light and dark color of text
are user configurable.

Resolves: #1978

thunze commented:
I just played around with the extension's source code to find a way to inherit the shell style in light mode. It's actually fairly simple: Just leave out the CSS that sets the `color` of app icons to `group-apps-label-font-color` / `group-apps-label-font-color-minimized`. So I think there could simply be an option that causes the extension to just leave out the CSS that overrides the font color.
